ansible/roles/metrics/files/process-cloudflare/process-cloudflare.js

```diff
@@ -0,0 +1,172 @@
+    if (winArch && winArch.indexOf('x64') >= 0) {
+      // could be 'x64' or 'win-x64'
+      return 'x64'
+    } else {
+      // could be 'win-x86' or ''
+      return 'x86'
+    }
+  }
+
+  return ''
+}
+
+const logTransformStream = new Transform({
+  writableObjectMode: true,
+  transform (chunk, encoding, callback) {
+    if (chunk.ClientRequestMethod !== 'GET' ||
+        chunk.EdgeResponseStatus < 200 ||
+        chunk.EdgeResponseStatus >= 300) {
+      return callback()
+    }
+
+    if (chunk.EdgeResponseBytes < 1024) { // unreasonably small for something we want to measure
+      return callback()
+    }
+
+    if (!extensionRe.test(chunk.ClientRequestPath)) { // not a file we care about
+      return callback()
+    }
+
+    const requestPath = chunk.ClientRequestPath.replace(/\/\/+/g, '/')
+    const uriMatch = requestPath.match(uriRe)
+    if (!uriMatch) { // what is this then?
+      return callback()
+    }
+
+    const path = uriMatch[1]
+    const pathVersion = uriMatch[4]
+    const file = uriMatch[5]
+    const winArch = uriMatch[6]
+    const fileVersion = uriMatch[8]
+    const fileType = uriMatch[9]
+
+    let version = ''
+    // version can come from the filename or the path, filename is best
+    // but it may not be there (e.g. node.exe) so fall back to path version
+    if (versionRe.test(fileVersion)) {
+      version = fileVersion
+    } else if (versionRe.test(pathVersion)) {
+      version = pathVersion
+    }
+
+    const os = determineOS(path, file, fileType)
+    const arch = determineArch(fileType, winArch, os)
+
+    const line = []
+    line.push(strftime('%Y-%m-%d', new Date(chunk.EdgeStartTimestamp / 1000 / 1000))) // date
+    line.push(chunk.ClientCountry.toUpperCase()) // country
+    line.push('') // state/province, derived from chunk.EdgeColoCode probably
+    line.push(chunk.ClientRequestPath) // URI
+    line.push(version) // version
+    line.push(os) // os
+    line.push(arch) // arch
+    line.push(chunk.EdgeResponseBytes)
+
+    this.push(`${line.join(',')}\n`)
+
+    callback()
+  }
+})
+
+
+exports.processLogs = async () => {
+  const file = data;
+  bucketName = file.bucket;
+  fileName = file.name;
+  console.log("DATA " + data);
+  console.log("BUCKET " + bucketName);
+  console.log("FILENAME " + fileName);
+
+  processedFile = fileName.split(".")[0];
+  processedFile = processedFile.split("_")[0].concat("_", processedFile.split("_")[1]); 
+
+  console.log("PROCESSEDFILENAME " + processedFile);
+
+  storage.bucket(bucketName).file(file.name).createReadStream()
+  .on('error', function(err) { console.error(err) })
+  .pipe(split2())
+  .pipe(jsonStream)
+  .pipe(logTransformStream)
+  .pipe(storage.bucket('processed-logs-nodejs').file(processedFile).createWriteStream({resumable: false, flags: 'a'})
+        .on("error", err => {
+          console.log("ERROR: >> ", err)
+        })
+        .on("finished", () => {
+        console.log("FINISHED")
+      }));
+}
```

Comment:
your function is an async arrow function, which I believe means it will have an implicit return.

The last function call you have here is to the bucket function and you create a readStream. AFAIK this is going to immediately return a readstream on the call to createReadStream, which will in turn resolve the async function promise, which will signal to GCP that the function is done and should be spun down (even if there is work to be done on the event loop).

You might want to switch to using a a non async function and explicitly call the callback argument when processing is done. Alternative you could utilize the promise API and use await to ensure that when the final function call inside the function returns the work is completed.